File: CODEVITA  TCS/VipCafe.py
import sys

def main():
    n = int(input())
    priorities = list(map(int, input().split()))
    k = int(input()) - 1  # Adjust to 0-based index
    
    served = 0
    while True:
        max_priority = max(priorities)
        max_index = priorities.index(max_priority)
        served += 1
        
        if max_index == k:
            break
        
        for i in range(max_index):
            priorities[i] += 1
        
        priorities = priorities[:max_index] + priorities[max_index+1:]
        if k > max_index:
            k -= 1
    
    print(served)

if __name__ == "__main__":
    main()

File: CODEVITA  TCS/VIPCafe.java
import java.util.Scanner;

public class VIPCafe {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] priorities = new int[n];
        for (int i = 0; i < n; i++) {
            priorities[i] = scanner.nextInt();
        }
        int k = scanner.nextInt();
        int served = 0;
        int index = k - 1;
        while (true) {
            int max = 0;
            int maxIndex = 0;
            for (int i = 0; i < n; i++) {
                if (priorities[i] > max) {
                    max = priorities[i];
                    maxIndex = i;
                }
            }
            served++;
            if (maxIndex == index) {
                break;
            }
            for (int i = 0; i < maxIndex; i++) {
                priorities[i]++;
            }
            for (int i = maxIndex; i < n - 1; i++) {
                priorities[i] = priorities[i + 1];
            }
            n--;
            if (index > maxIndex) {
                index--;
            }
        }
        System.out.println(served);
        scanner.close();
    }
}